kafka/consumer.go

```diff
@@ -165,9 +165,9 @@ func (c *Consumer) Close() error {
 	if !c.closed.CompareAndSwap(false, true) {
```

Comment:
FYI: It was introduced in [this PR](https://github.com/elastic/apm-queue/pull/173) with the expectation that we will return nil if the client is already closed. Maybe what we need is `defer c.closed.Swap(true)`(??)

Reply:
:ok_hand:  I think we should generally avoid atomics -- channels are typically more appropriate when trying to communicate between goroutines, to avoid race conditions. If we need an atomic, then we can indeed `defer c.closed.Store(true)`.

Reply:
> Maybe what we need is defer c.closed.Swap(true) if we just want to know if Close succeeded or not

I _think_ this in itself will not be enough, sorry for the confusion. The reason is that after checking with `c.closed.Load()` and before acquiring the read lock, the method `Close` can be called. This I assume will cause `fetch` to fail with a different error than expected `context.Canceled`. I think @axw's suggestion to use channels in this case and interrupt using the channel inside the read lock is better.